domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmAddHandler.java

```diff
@@ -779,7 +781,15 @@ private void addSecretService(OperationContext context, ModelNode secret, String
         ServiceName secretServiceName = SecretIdentityService.ServiceUtil.createServiceName(realmName);
 
         ModelNode resolvedValueNode = SecretServerIdentityResourceDefinition.VALUE.resolveModelAttribute(context, secret);
-        boolean base64 = secret.get(SecretServerIdentityResourceDefinition.VALUE.getName()).getType() != ModelType.EXPRESSION;
+
+        boolean base64=true;
+        if(secret.get(SecretServerIdentityResourceDefinition.VALUE.getName()).getType() == ModelType.EXPRESSION) {
+           String pattern = "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$";
```

Comment:
I think the remainder of the discussion needs to continue but for any place where we add a regular expression can we please add a comment block documenting how the pattern is constructed and if relevant why.

Retrospectively working on a regular expression without the context that led to it's development can be a very painful process.